app/org/sagebionetworks/bridge/validators/UploadSchemaValidator.java

```diff
@@ -101,16 +107,50 @@ public void validate(Object target, Errors errors) {
                             errors.rejectValue("type", "is required");
                         }
 
-                        List<String> multiChoiceAnswerList = fieldDef.getMultiChoiceAnswerList();
-                        if (fieldType == UploadFieldType.MULTI_CHOICE && (multiChoiceAnswerList == null ||
-                                multiChoiceAnswerList.isEmpty())) {
-                            errors.rejectValue("multiChoiceAnswerList", "must be specified for MULTI_CHOICE field "
-                                    + fieldName);
+                        if (fieldType == UploadFieldType.MULTI_CHOICE) {
+                            List<String> multiChoiceAnswerList = fieldDef.getMultiChoiceAnswerList();
+                            if (multiChoiceAnswerList == null || multiChoiceAnswerList.isEmpty()) {
+                                errors.rejectValue("multiChoiceAnswerList", "must be specified for MULTI_CHOICE field "
+                                        + fieldName);
+                            } else {
+                                // Multi-Choice fields create extra "sub-field" columns, and we need to check for
+                                // potential name collisions.
+                                //noinspection Convert2streamapi
+                                for (String oneAnswer : multiChoiceAnswerList) {
+                                    fieldNameList.add(fieldName + MULTI_CHOICE_FIELD_SEPARATOR + oneAnswer);
+                                }
+                            }
+
+                            if (Boolean.TRUE.equals(fieldDef.getAllowOtherChoices())) {
+                                // Similarly, there's an "other" field.
+                                fieldNameList.add(fieldName + OTHER_CHOICE_FIELD_SUFFIX);
+                            }
+                        } else if (fieldType == UploadFieldType.TIMESTAMP) {
+                            // Timestamp fields also generate a second subfield for timezone. Need to check for name
+                            // collisions here too.
+                            fieldNameList.add(fieldName + TIME_ZONE_FIELD_SUFFIX);
                         }
 
                         errors.popNestedPath();
                     }
                 }
+
+                // Check for duplicate field names. Dupe set is a tree set so our error messages are in a predictable
+                // alphabetical order.
+                Set<String> seenFieldNameSet = new HashSet<>();
+                Set<String> dupeFieldNameSet = new TreeSet<>();
+                for (String oneFieldName : fieldNameList) {
+                    if (seenFieldNameSet.contains(oneFieldName)) {
+                        dupeFieldNameSet.add(oneFieldName);
+                    } else {
+                        seenFieldNameSet.add(oneFieldName);
+                    }
+                }
```

Comment:
I believe so. This is because sanitization is happening in the builder, and validation can only be called on a built object.

I'll add a test to verify this.